src/generic/hooks/useOverflowControl.js

```diff
@@ -0,0 +1,27 @@
+import { useEffect } from 'react';
+
+const useOverflowControl = (targetSelector) => {
```

Comment:
Docstrings are required here. Not clear on the first glance what this hook does.

Reply:
Added